### DISCUSSION TITLE
Projective sampling under orthographic projection

### DISCUSSION THREAD

**Discussion by ejtwe** [mitsuba-renderer/mitsuba3#1160] [Q&A]
How can projective sampling be extended to compute differentiable gradients under orthographic projection, and to optimize geometry? This should also be an interesting question.

**Comment by njroussel** [comment]
Hi @ejtwe 

This is definitely possible and is slightly simpler for a few things. We actually had an implementation for it some time ago, but removed for the release.

The interfaces required for this are different from the ones of  a `perspective`-style camera. Consider a ray starting at your sensor, and that is tangential to your geometry (it's on its directly visible silhouette). In order to move to either side of the silhouette/discontinuity, you either need to move your ray origin with an `orthographic` sensor, or the ray direction with a `perspective` sensor. The required information by each senor type is different. There are also some jacobians that would also need to be changed because of this difference.

Getting this implemented nicely in a unified interface isn't straightforward. In addition, we didn't really have the time to revisit this, hence why it hasn't been done yet.
